File: 백준/Silver/10773. 제로/제로.java
import java.util.Scanner;
import java.util.Stack;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        Stack<Integer> stack = new Stack<Integer>();

        int K = sc.nextInt();

        for(int i = 0; i < K; i++) {
            int num = sc.nextInt();

            if(num == 0) {
                stack.pop();
            }
            else {
                stack.push(num);
            }
        }
        int sum = 0;
        for (Integer n : stack) {
            sum += n;
        }
        System.out.println(sum);
    }
}

File: 백준/Silver/10773. 제로/제로.py
class Stack:
    def __init__(self, size):
        self.top=-1
        self.stack=[]
        self.size=size

    def push(self,item):
        if self.top < (self.size-1):
            self.top += 1
            self.stack.append(item)
        else: return
        
    def pop(self):
        if self.top > -1:
            self.top -=1
            return self.stack.pop()
        else: return

n=int(input())
s=Stack(n)
sum=0
for i in range(n):
    num=int(input())
    s.push(num)
    sum+=num
    if num==0:
        s.pop()
        sum-=s.pop()
print(sum)